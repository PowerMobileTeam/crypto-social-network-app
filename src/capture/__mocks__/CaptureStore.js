import AttachmentStore from '../../common/stores/__mocks__/AttachmentStore';

const mock = jest.fn().mockImplementation(() => {

  return {
    setPosting: jest.fn(),
    reset: jest.fn(),
    post: jest.fn(),
    loadThirdPartySocialNetworkStatus: jest.fn(),
    attachment: new AttachmentStore(),
    isPosting: false,
    socialNetworks: {
      facebook: false,
      twitter: false,
    }
  };
});

export default mock;
