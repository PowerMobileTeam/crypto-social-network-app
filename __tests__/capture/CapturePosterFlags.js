import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { whenWithTimeout } from 'mobx-utils';
import { useStrict } from 'mobx';

import LicensePicker from '../../src/common/components/LicensePicker';
import CaptureStore from  '../../src/capture/CaptureStore';
import CapturePosterFlags from '../../src/capture/CapturePosterFlags';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('../../src/capture/CaptureStore');
jest.mock('../../src/common/components/LicensePicker', () => 'LicensePicker');

defaultState = {
  mature: false,
  share: false,
  lock: false
}

/**
 * Test render with  value
 */
const testRenderWithValue = (value) => {
  fn = () => null;

  const store = new CaptureStore();

  state = {
    mature: value,
    share: value,
    lock: value
  }

  const preview = renderer.create(
    <CapturePosterFlags.wrappedComponent
        capture={store}
        matureValue={state.mature}
        shareValue={state.share}
        lockValue={state.lock}
        onMature={fn}
        onShare={fn}
        onLocking={fn}
      />
  ).toJSON();
  expect(preview).toMatchSnapshot();
}

// turn off mobx strict mode
useStrict(false);

/**
 * Tests
 */
describe('cature poster flags component', () => {

  it('should renders correctly for props false', () => {
    testRenderWithValue(false)
  });

  it('should renders correctly for props true', () => {
    testRenderWithValue(true)
  });


  it('should load third party social networks status on will mount', () => {

    const store = new CaptureStore();

    store.loadThirdPartySocialNetworkStatus.mockClear();

    const wrapper = shallow(
      <CapturePosterFlags.wrappedComponent
        capture={store}
        matureValue={defaultState.mature}
        shareValue={defaultState.share}
        lockValue={defaultState.lock}
        onMature={fn}
        onShare={fn}
        onLocking={fn}
      />
    );


    expect(store.loadThirdPartySocialNetworkStatus).toHaveBeenCalled();
  });

  it('should show license picker if it has an attachment', () => {

    const store = new CaptureStore();

    //store.attachment.hasAttachment = true;

    const wrapper = shallow(
      <CapturePosterFlags.wrappedComponent
        capture={store}
        matureValue={defaultState.mature}
        shareValue={defaultState.share}
        lockValue={defaultState.lock}
        onMature={fn}
        onShare={fn}
        onLocking={fn}
      />
    );
    let picker;
    // find License picker
    picker = wrapper.dive().find(LicensePicker);

    // check there is no license picker
    expect(picker.length).toEqual(0);

    store.attachment.hasAttachment = true;

    // update the component
    wrapper.update();

    picker = wrapper.dive().find(LicensePicker);

    // check there is 1 license picker
    expect(picker.length).toEqual(1);
  });

});
