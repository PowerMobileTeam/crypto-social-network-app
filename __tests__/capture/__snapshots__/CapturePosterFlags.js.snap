// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cature poster flags component should renders correctly for props false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "flexDirection": "row",
      "marginBottom": 5,
      "marginRight": 8,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#777777",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#777777",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#777777",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      enabled={true}
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexGrow": 1,
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 1,
                  "marginLeft": 10,
                }
              }
            >
              SHARE
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  Object {
                    "alignSelf": "flex-end",
                    "marginRight": 10,
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#cccccc",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#777777",
                    "flexGrow": 1,
                    "fontFamily": "Roboto",
                    "fontSize": 13,
                    "letterSpacing": 1,
                    "marginLeft": 6,
                  },
                  undefined,
                ]
              }
            >
              FACEBOOK
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#cccccc",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#777777",
                    "flexGrow": 1,
                    "fontFamily": "Roboto",
                    "fontSize": 13,
                    "letterSpacing": 1,
                    "marginLeft": 6,
                  },
                  undefined,
                ]
              }
            >
              TWITTER
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      enabled={true}
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexGrow": 1,
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 1,
                  "marginLeft": 10,
                }
              }
            >
              LOCK BY WIRE
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  Object {
                    "alignSelf": "flex-end",
                    "marginRight": 10,
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 15,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                keyboardType="numeric"
                onChangeText={[Function]}
                style={
                  Object {
                    "backgroundColor": "#eee",
                    "borderRadius": 3,
                    "color": "#666",
                    "flexGrow": 1,
                    "fontSize": 16,
                    "fontWeight": "300",
                    "marginRight": 3,
                    "padding": 5,
                    "textAlign": "right",
                  }
                }
                value="0"
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontFamily": "Roboto",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 1,
                  }
                }
              >
                TOKENS
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 15,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "borderColor": "#ececec",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "padding": 8,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    Object {
                      "borderColor": "#4690D6",
                    },
                    false,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "paddingLeft": 5,
                      },
                      Object {
                        "paddingRight": 5,
                      },
                      Object {
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "color": "#4690D6",
                      },
                      false,
                    ]
                  }
                >
                  Done
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`cature poster flags component should renders correctly for props true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "flexDirection": "row",
      "marginBottom": 5,
      "marginRight": 8,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#f44336",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#f44336",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "600",
          "marginLeft": 3,
        }
      }
    >
      MATURE
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#777777",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRightWidth": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#777777",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      enabled={true}
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexGrow": 1,
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 1,
                  "marginLeft": 10,
                }
              }
            >
              SHARE
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  Object {
                    "alignSelf": "flex-end",
                    "marginRight": 10,
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#cccccc",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#777777",
                    "flexGrow": 1,
                    "fontFamily": "Roboto",
                    "fontSize": 13,
                    "letterSpacing": 1,
                    "marginLeft": 6,
                  },
                  undefined,
                ]
              }
            >
              FACEBOOK
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#cccccc",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#777777",
                    "flexGrow": 1,
                    "fontFamily": "Roboto",
                    "fontSize": 13,
                    "letterSpacing": 1,
                    "marginLeft": 6,
                  },
                  undefined,
                ]
              }
            >
              TWITTER
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      enabled={true}
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexGrow": 1,
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "letterSpacing": 1,
                  "marginLeft": 10,
                }
              }
            >
              LOCK BY WIRE
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 28,
                  },
                  Object {
                    "alignSelf": "flex-end",
                    "marginRight": 10,
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 15,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                keyboardType="numeric"
                onChangeText={[Function]}
                style={
                  Object {
                    "backgroundColor": "#eee",
                    "borderRadius": 3,
                    "color": "#666",
                    "flexGrow": 1,
                    "fontSize": 16,
                    "fontWeight": "300",
                    "marginRight": 3,
                    "padding": 5,
                    "textAlign": "right",
                  }
                }
                value="0"
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#666",
                    "fontFamily": "Roboto",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 1,
                  }
                }
              >
                TOKENS
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 15,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "borderColor": "#ececec",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "padding": 8,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    Object {
                      "borderColor": "#4690D6",
                    },
                    false,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "paddingLeft": 5,
                      },
                      Object {
                        "paddingRight": 5,
                      },
                      Object {
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "color": "#4690D6",
                      },
                      false,
                    ]
                  }
                >
                  Done
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
